### ISSUE TITLE
shared_with_groups returns different types for Groups and Projects

### ISSUE THREAD

**Issue by Rein142** [jdalrymple/gitbeaker#3579]
**Description**
I am trying to implement a function in React (typescript) that compares the properties of shared_with_groups on a Group or Project with ones that a user wants to add to said resource. But I am running into the fact that on Groups, calling on the shared_with_groups attribute, gitbeaker returns this as a ProjectSchema array, whereas on Projects it returns this as a string array. Is this an intended feature? I don't see how the following output makes sense in a ProjectSchema:
```
"shared_with_groups": [
  {
    "group_id": <id>,
    "group_name": "<name>",
    "group_full_path": "<path>",
    "group_access_level": <access_level>,
    "expires_at": <expiration_date>
  }
]
```
Can someone tell me whether this is intended or not, and if it is, why exactly? 

I call upon these properties like this:
```
const groupInfo = await this.gitlabApi.api.Groups.show(<group_id>)
const groupSharedWithGroups = groupInfo.shared_with_groups

const projectInfo = await this.gitlabApi.api.Projects.show(<project_id>)
const projectSharedWithGroups = projectInfo.shared_with_groups
```

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

**Comment by jdalrymple** [comment]
Sounds like a typo, but ill double check it and follow up

**Comment by Rein142** [comment]
Looking at the fix it changed the string array return type for the ProjectSchema, but within the ExpandedGroupSchema it still returns a ProjectSchema array, is this intended then?

**Comment by jdalrymple** [comment]
> Looking at the fix it changed the string array return type for the ProjectSchema, but within the ExpandedGroupSchema it still returns a ProjectSchema array, is this intended then?

It isnt, just overlooked.

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`40.1.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/40.1.0) :rocket:

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`40.1.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/40.1.0) :rocket:
